### PR TITLE
Nodata values not available

### LINKED ISSUE #82: Nodata values not available when creating dataset from array
Creating an `Image.dataset` from a numpy array produces a rasterio dataset with the following metadata:

```
meta = {
    "driver": "GTiff",
    "dtype": self.__arr.dtype,
    "nodata": nodata,
    "height": self.__arr.shape[-2],
    "width": self.__arr.shape[-1],
    "count": self.__arr.shape[0],
    "crs": crs,
    "transform": transform,
}
```

The instance `nodata` is given as an argument of the method: `__update_dataset(self, crs, transform, nodata=None)`. However, `nodata` is not handed upon function call `self.__update_dataset(crs, transform)` which leaves the new dataset with no nodata value.

Suggest to introduce a nodata value when creating datasets from arrays such as `self.__update_dataset(crs, transform, nodata=nodata)`. This requires handing the nodata value to the instantiation of the Image class `__init__(self, data, dimorder="first", crs=None, transform=None, nodata=None)` if usage is desired.

### PR DESCRIPTION
**Pull request by BoehnkeC**
Closes #82.

### CODE REVIEW

File: ukis_pysat/raster.py [hunk 1]
```diff
@@ -30,12 +30,13 @@ class Image:
...
     da_arr = None
 
-    def __init__(self, data, dimorder="first", crs=None, transform=None):
+    def __init__(self, data, dimorder="first", crs=None, transform=None, nodata=None):
         """
         :param data: rasterio.io.DatasetReader or path to raster or np.ndarray of shape (bands, rows, columns)
         :param dimorder: Order of channels or bands 'first' or 'last' (default: 'first')
         :param crs: Coordinate reference system used when creating form array. If 'data' is np.ndarray this is required (default: None)
         :param transform: Affine transformation mapping the pixel space to geographic space. If 'data' is np.ndarray this is required (default:None)
+        :param nodata: nodata value, optional
```
Comment (fwfichtner) [inline ukis_pysat/raster.py:39]:
```suggestion
        :param nodata: nodata value used when creating form array, optional (default: None)

```

File: tests/test_raster.py [hunk 2]
```diff
@@ -109,6 +109,17 @@ def test_arr(self):
...
 
+    def test_arr_nodata(self):
+        array = np.ones((1, 385, 502))
+        array[0, 0, 0] = 0
+
+        img_nodata = Image(
+            array, crs=self.img.dataset.crs, transform=self.img.dataset.transform, nodata=0
+        )
+
+        self.assertEqual(img_nodata.dataset.nodata, 0.0)
```
Comment (fwfichtner) [inline tests/test_raster.py:120]: Why are you mixing float and int? You also don't even have to have nodata in an array to set it, right?
Please also check with `nodatavals`.

**Review by fwfichtner — APPROVED** [review]
Please check minor details, generally good.

File: tests/test_raster.py (hunk 2 above)
Comment (BoehnkeC) [reply tests/test_raster.py:120]: I did not check with `nodatavals` as I think it requires a separate discussions on how to treat nodata values across different bands.

Comment (BoehnkeC) [reply tests/test_raster.py:120]: You are right: it is not required to have the actual nodata value in the array. However, I included it for the sake of completeness.

Comment (BoehnkeC) [reply tests/test_raster.py]: > Why are you mixing float and int?
The returned nodata value is of type float and it was intuitive for me to test for float but rasterio does not care about the datatype of the number. Changed as suggested by you.

**Review by fwfichtner — APPROVED** [review]
Well done.

**Pull request merged**